sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/Sampler.java

```diff
@@ -88,5 +89,14 @@ SamplingResult shouldSample(
      *     {@link Decision#RECORD_AND_SAMPLE}.
      */
     Attributes getAttributes();
+
+    /**
+     * Return the {@link TraceState} that should be associated with the resulting Span. This may be
+     * the same {@link TraceState} that was provided originally, or an updated one.
+     *
+     * <p>Note: if a null is returned, the TraceState will be reset to {@link
```

Comment:
```suggestion
     * <p>Note: if {@code null} is returned, the TraceState will be reset to {@link
```

Or I might remove this, this is not a nullable method so we don't want to indicate null is allowed. We just happen to have a default to prevent crashing in case that happens.

Reply:
Indeed, should we require non-null return value?

Reply:
The spec explicitly specifies the behavior for the `null` return, but I wouldn't have a problem making this non-nullable. It's easy enough for samplers to explicitly return `TraceState.getInvalid()` if that's what they want.

Reply:
Sorry..my comment above is wrong. I mis-read "empty" as "null" in the spec.